app/src/main/java/com/example/peopleloader/LoadActivity.java

```diff
@@ -1,17 +1,182 @@
+import android.support.v4.content.ContextCompat;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
 
 /*
  *
  */
 
 public class LoadActivity extends AppCompatActivity {
+    private static final int REQUEST_INTERNET_THEN_ADD = 34624;
+    @BindView (R.id.url_entry) protected EditText mUrlEntry;
+    @BindView (R.id.progressBar) protected ProgressBar mProgressBar;
+    @BindView (R.id.view_button) protected Button mViewButton;
+    @BindView (R.id.add_button) protected Button mAddButton;
+    private DatabaseHelper mDatabaseHelper;
+    private LoadPeopleTask mLoadPeopleTask = null;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_load);
+        ButterKnife.bind(this);
+        mDatabaseHelper = new DatabaseHelper(this);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        //Don't need to worry about type of permission, because we only ever request
+        //Internet permissions for this app
+        switch (requestCode) {
+            case REQUEST_INTERNET_THEN_ADD:
+                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    add();
+                } else {
+                    Toast.makeText(this, R.string.must_enable_permission, Toast.LENGTH_SHORT).show();
+                }
+        }
+    }
+
+    @OnClick (R.id.add_button)
+    public void add() {
+        //If we have the correct permissions AND we are not already loading people
+        if (internetPermissionEnabled(this) && mLoadPeopleTask == null ) {
+            try {
+                URL url = new URL(mUrlEntry.getText().toString());
+                mLoadPeopleTask = new LoadPeopleTask();
+                mLoadPeopleTask.execute(url);
+            } catch (MalformedURLException e) {
+                Toast.makeText(this, R.string.url_invalid, Toast.LENGTH_SHORT).show();
+            }
+        } else if (!internetPermissionEnabled(this)) {
+            resolveInternetPermission(REQUEST_INTERNET_THEN_ADD, this);
+        }
+
+    }
+
+    @OnClick (R.id.view_button)
+    public void viewPeople() {
+        //Go to a new activity to load people from the database
+    }
+
+    boolean internetPermissionEnabled(Context context) {
+        return (ContextCompat.checkSelfPermission(context, Manifest.permission.INTERNET) ==

```

Comment:
You don't need to check for the `INTERNET` permission.  `INTERNET` permission is part of the  [normal android permission](https://developer.android.com/guide/topics/permissions/normal-permissions.html) which are granted at install time if specified in the `AndroidManifest.xml` file.

The other class of permissions is protected permissions. Those you have to specify in `AndroidManifest.xml` and ask for them at run time.